Video Texture set to repeat is clamped

kkinnunen-apple opened the pull request:
#### c340b7473bca7f6b64867669aeca7ebc169531a8
<pre>
Video Texture set to repeat is clamped
<a href="https://bugs.webkit.org/show_bug.cgi?id=248543">https://bugs.webkit.org/show_bug.cgi?id=248543</a>
rdar://problem/102912865

Reviewed by Kenneth Russell.

The texImage2D fast-path would change the target texture wrap and min/mag properties.
The FBO where the target texture will be bound is complete regardless
of the properties.

* LayoutTests/webgl/resources/pot-video.mp4: Added.
* LayoutTests/webgl/tex-2d-video-no-change-wrap-expected.txt: Added.
* LayoutTests/webgl/tex-2d-video-no-change-wrap.html: Added.
* Source/WebCore/platform/graphics/cv/GraphicsContextGLCVCocoa.cpp:
(WebCore::GraphicsContextGLCVCocoa::copyVideoSampleToTexture):

Canonical link: <a href="https://commits.webkit.org/257365@main">https://commits.webkit.org/257365@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/20898c7a78c56b9be871176f025b73ba9a219344

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98566 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/7776 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/31689 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107988 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168252 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102515 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8286 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85154 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91100 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/105973 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104248 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6283 "Found 2 new test failures: css3/supports-dom-api.html, fast/text/text-edge-no-half-leading-simple.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89837 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33272 "Found 1 new test failure: css3/supports-dom-api.html (failure)") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88101 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21196 "Found 4 new test failures: css3/supports-dom-api.html, fast/text/text-edge-no-half-leading-simple.html, fast/text/text-edge-no-half-leading-with-line-height-simple.html, fast/text/text-edge-with-margin-padding-border-simple.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76211 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1709 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22725 "Found 3 new test failures: css3/supports-dom-api.html, fast/text/text-edge-no-half-leading-with-line-height-simple.html, fast/text/text-edge-with-margin-padding-border-simple.html (failure)") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1622 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45223 "Found 3 new test failures: fast/text/text-edge-with-margin-padding-border-simple.html, imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/modal-dialog-selection.html, media/video-inaccurate-duration-ended.html (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5055 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6560 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42150 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3009 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kenrussell reviewed (approved):
Great find and fix, and great job on the test. r+

djg commented on LayoutTests/webgl/tex-2d-video-no-change-wrap.html:
```diff
@@ -0,0 +1,86 @@
+var wtu = WebGLTestUtils;
+var gl;
+// Video with:
+// ffmpeg -f lavfi -i color=c=red@0.2:d=1:s=512x512:r=25 -pix_fmt yuv420p -filter_complex "drawbox=t=fill:x=0:y=0:w=256:h=256:color=0x102030, drawbox=t=fill:x=255:y=0:w=256:h=256:color=0x405060, drawbox=t=fill:x=0:y=255:w=256:h=256:color=0x708090, drawbox=t=fill:x=255:y=255:w=256:h=256:color=0xA0B0C0" -colorspace bt709 -color_primaries bt709 -color_trc iec61966_2_1 pot-video.mp4  -y
```
Is this important to document?

kenrussell replied:
Yes. When revisiting these tests, often years later and not by the original author, it's incredibly useful to know how the resources were produced.

djg reviewed (approved):
LGTM

kkinnunen-apple committed:
Video Texture set to repeat is clamped
https://bugs.webkit.org/show_bug.cgi?id=248543
rdar://problem/102912865

Reviewed by Kenneth Russell.

The texImage2D fast-path would change the target texture wrap and min/mag properties.
The FBO where the target texture will be bound is complete regardless
of the properties.

* LayoutTests/webgl/resources/pot-video.mp4: Added.
* LayoutTests/webgl/tex-2d-video-no-change-wrap-expected.txt: Added.
* LayoutTests/webgl/tex-2d-video-no-change-wrap.html: Added.
* Source/WebCore/platform/graphics/cv/GraphicsContextGLCVCocoa.cpp:
(WebCore::GraphicsContextGLCVCocoa::copyVideoSampleToTexture):

Canonical link: https://commits.webkit.org/257365@main
